hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -862,6 +892,12 @@ public void notifyTermIndexUpdated(long term, long index) {
     return future;
   }
 
+  @Override
+  public void notifyNotLeader(Collection<TransactionContext> pendingEntries) {
+    // once the leader steps down , clear the cache
+    evictStateMachineCache();
```

Comment:
By default we keep the state machine data in both follower and leader because new leader might need to resend the data.

Reply:
Th new leader will resend only if a follower doesn't have the log entry itself even if the entry exists in the stateMachineDataCache.

 A follower by default doesn't have any entry in the stateMachine cache but only the leader.

Reply:
I see. This is not an issue then.